spec/forms/legal_aid_applications/restrictions_form_spec.rb

```diff
@@ -0,0 +1,81 @@
+require 'rails_helper'
+
+RSpec.describe LegalAidApplications::RestrictionsForm, type: :form do
+  let(:application) { create :legal_aid_application, :with_applicant }
+  let(:params) do
+    {
+      has_restrictions: has_restrictions,
+      restrictions_details: restrictions_details
+    }
+  end
+  let(:restrictions_details) { 'xxxxx' }
+  let(:has_restrictions) { 'true' }
+  let(:form_params) { params.merge(model: application) }
```

Comment:
You could also pass a `mode`. To make sure it is used like in the controller.
`let(:mode) { %i[provider citizen].sample }`